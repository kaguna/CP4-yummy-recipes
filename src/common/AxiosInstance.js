import axios from 'axios';

/**
 * Component for handling Api calls without duplicating in each file.
 */
const axiosInstance = axios.create({
  baseURL: 'http://localhost:5000',
  headers: { 'Content-Type': 'application/json', 'x-access-token': `${localStorage.getItem('token')}` },
});

axiosInstance.interceptors.request.use((config) => {
  if (localStorage.getItem('token') && config.headers['x-access-token'] === 'null') {
    config.headers['x-access-token'] = `${localStorage.getItem('token')}`;
  }
  return config;
});

export default axiosInstance;
