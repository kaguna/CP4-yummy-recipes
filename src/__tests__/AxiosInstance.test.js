import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AxiosInstance from '../common/AxiosInstance';

describe('AxiosInstance component', () => {
  const wrapper = shallow(<AxiosInstance />);

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
