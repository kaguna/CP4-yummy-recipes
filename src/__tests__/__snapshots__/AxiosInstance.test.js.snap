// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AxiosInstance component renders properly 1`] = `undefined`;
